sig-and-wg/sig-core/proposals/update-code-and-chart-in-one-pr.md

```diff
@@ -0,0 +1,111 @@
+- The `kyma` repository has a transparent commit history and commits that only bump image versions are rare.
+- Run Kyma integration tests with the modified component within the same pull request status checks.
+
+
+## Proposed solution
+
+The solution assumes using images built on pull requests on the master branch.
+
+Let assume that I am working on PR-1234. When I modify `componentA`, presubmit jobs build Docker images. 
+The image has the same tag as the pull request number.
+A developer can use this tag as a version in the `values.yaml` file:
+```
+component_a:
+  dir: pr/
+  version: PR-1234
+```
+
+With this approach, Prow executes component and integration jobs for the same PR. We have to ensure the proper order of jobs
+to build components before using them in integration jobs. To achieve that, an additional step is required at the beginning of 
+every integration job that waits for all dependent jobs. See [this](#guard-integration-jobs) section for more details.
+
+In the beginning, the described approach can be optional, which means that a developer can decide whether he updates code and chart in the same PR or not. 
+As a next step, we can introduce a job that checks if a version of the chart is updated in the PR. See
+[this](#job-enforcing-changes-in-one-pr) section for more information.
+
+In case there are two PRs that change the same component, there will be a merge conflict for the second PR because
+it will modify the same line in the `values.yaml` file. Such approach ensures that the `master` branch contains changes from both PRs.
+
+### Developers workflow
+A developer is working on an issue that requires changes in the `helm-broker` component.
+- Create a branch.
+- Introduce changes to the `helm-broker`. Write or update unit tests.
+- In order to test introduced changes in Kyma, create a Pull Request. Pull request has number `1234`. Because only `helm-broker` code was modified, 
+the `pre-master-kyma-component-helm-broker` job is executed. All other jobs are skipped.
+If the job is successful, the component's image is published: `eu.grc.io/kyma-project/pr/helm-broker:PR-1234`.
+- A developer tests changes locally on a minikube. In order to use the newly created image, he edits `kyma-project/kyma/resources/helm-broker/values.yaml` file:
+```
+global:
+
+  helm_broker:
+    dir: pr/
+    version: PR-1234
+```
+If everything is fine, commit changes made to `values.yaml` file.
+- The following Prowjobs are triggered:
+    - `pre-master-kyma-component-helm-broker` - even though component was not changed in this commit, Prow triggers all jobs
+according to the changes introduced in the PR.
+    - all integration jobs: `pre-master-kyma-integration`, `pre-master-kyma-gke-integration`, `pre-master-kyma-gke-upgrade` etc. 
+All those jobs are triggered because the file in `resources` directory was modified. All integration jobs have a Guard step that waits
```

Comment:
```suggestion
All these jobs are triggered because the file in the `resources` directory was modified. All integration jobs have a Guard step that waits
```